controllers/terminal_controller.go

```diff
@@ -506,7 +506,7 @@ func (r *TerminalReconciler) reconcileTerminal(ctx context.Context, targetClient
 }
 
 func (r *TerminalReconciler) createOrUpdateAttachPodSecret(ctx context.Context, hostClientSet *gardenclient.ClientSet, t *extensionsv1alpha1.Terminal, labelSet *labels.Set, annotationSet *utils.Set) error {
-	if t.Spec.Host.TemporaryNamespace {
+	if t.Spec.Host.TemporaryNamespace != nil && *t.Spec.Host.TemporaryNamespace {
```

Comment:
why not `utils.NotNilAndTrue` in this case?

Reply:
✅  I have also switched to the `BoolDeref` function